build/scripts/test-build-correctness.ps1

```diff
@@ -36,7 +36,31 @@ try {
     # Verify the state of our generated syntax files
     Write-Host "Checking generated compiler files"
     Exec-Block { & (Join-Path $PSScriptRoot "generate-compiler-code.ps1") -test }
+    Write-Host ""
+    
+    # Verfiy the state of creating run settings for optprof
+    Write-Host "Checking run generation for optprof"
+
+    # set environment variables
+    if (-not (Test-Path env:SYSTEM_TEAMPROJECT)) { $env:SYSTEM_TEAMPROJECT = "DevDiv" }
+    if (-not (Test-Path env:BUILD_REPOSITORY_NAME)) { $env:BUILD_REPOSITORY_NAME = "dotnet/roslyn" }
+    if (-not (Test-Path env:BUILD_SOURCEBRANCHNAME)) { $env:BUILD_SOURCEBRANCHNAME = "test" }
+    if (-not (Test-Path env:BUILD_BUILDID)) { $env:BUILD_BUILDID = "42.42.42.42" }
+    if (-not (Test-Path env:BUILD_SOURCESDIRECTORY)) { $env:BUILD_SOURCESDIRECTORY = $RepoRoot }
+    if (-not (Test-Path env:BUILD_STAGINGDIRECTORY)) { $env:BUILD_STAGINGDIRECTORY = $configDir }
```

Comment:
Why only conditionally setting these variables? Isn't it simpler to juts always set them?

Reply:
They will be set in the ADO build so this is for running them on your local machine